### PR TITLE
Add partial 2D dam break in ex1.c

### PR DESCRIPTION
**Pull request by donghuix**
Follow partial 2D dam break on a flat surface from Section 6.1 of https://www.sciencedirect.com/science/article/pii/S0045782509002527#fd1

### CODE REVIEW

File: swe_roe/show_ex1.m [hunk 1]
```diff
@@ -2,11 +2,15 @@
...
 dof = 3;
-Nx  = 51;
-Ny  = 51;
-dt  = 0.1;
+Nx  = 200;
+Ny  = 200;
+dt  = 0.04;
+
+x = 1 : 200;
+y = 1 : 200;
```
Comment (bishtgautam) [inline swe_roe/show_ex1.m]: How about letting `ex1.c` save this information and then having the matlab code read it?

File: swe_roe/ex1.c [hunk 2]
```diff
@@ -498,6 +523,8 @@ int main(int argc, char **argv)
    		PetscCall(PetscOptionsInt("-Nt","Number of time steps","",user->Nt,&user->Nt,NULL));
 	    PetscCall(PetscOptionsReal("-hx","dx","",user->hx,&user->hx,NULL));
 	    PetscCall(PetscOptionsReal("-hy","dy","",user->hy,&user->hy,NULL));
+        PetscCall(PetscOptionsReal("-hu","hu","",user->hu,&user->hu,NULL));
+        PetscCall(PetscOptionsReal("-hd","hd","",user->hd,&user->hd,NULL));
```
Comment (bishtgautam) [inline swe_roe/ex1.c:632]: How about adding a check to ensure the heights provided by a user are non-negative?

File: swe_roe/ex1.c [hunk 3]
```diff
@@ -116,9 +117,31 @@ PetscErrorCode Add_Buildings(User user)
...
+    x x x x x x x x x x x
+    
+    */
     for (j = user->ys; j < user->ys + user->ym; j = j + 1) {
         for (i = user->xs; i < user->xs + user->xm; i = i + 1) {
-            if (i > xmin && i < xmax && j > ymin && j < ymax) {
+            if (i < 30 && j >= 95 && j < 105) {
+                b_ptr[j][i][0] = 1.;
+            }
+            else if (i >= 105 && j >= 95 && j < 105) {
```
Comment (bishtgautam) [inline swe_roe/ex1.c]: I believe the location of the wall is assuming `user->hx = user->hy = 1.0`. The code should be modified to explicitly take into account `hx` and `hy`, so the wall remains in the same location when the grid spacing is modified.

**Review by bishtgautam — CHANGES_REQUESTED** [review]
Few minor comments

File: swe_roe/ex1.c (hunk 3 above)
Comment (bishtgautam) [reply swe_roe/ex1.c]: @donghuix Let me know if you think addressing this comment requires significant work that you would prefer to fix in a future PR.

**Comment by donghuix** [comment]
@bishtgautam, I updated the code based on your comments in above commit. Please let me know if you have other comments.

**Review by bishtgautam — APPROVED** [review]

**Pull request merged**